LEGACY_SSE_EMULATION.md

```diff
@@ -0,0 +1,512 @@
+# Rationale
+
+The casper node 2.x produces a different set of SSE events than the 1.x ones. Also, 1.x nodes used 3 sse endpoints (`/events/sigs`, `/events/deploys`, `/events/main`), while 2.x node exposed all sse events under one firehose endpoint (`/events`).
+
+Generally the changes in 2.x regarding SSE are backwards incompatible to some extend. To harness all the details and collect all the data clients should adapt the new SSE API. However if some clients are not ready or have no need to adapt to the new SSE API, they can use the legacy SSE emulation.
+
+SSE emulation is by default turned off, the instruction on how to enable it is in the (main README.md)[./README.md] file.
+
+**BEFORE YOU ENABLE LEGACY SSE EMULATION** please consider the following:
+* The legacy SSE emulation is a temporary solution and can be removed in a future major release.
+* The legacy SSE emulation is not a 1:1 mapping of the 2.x events to 1.x events. Some events will be omitted, some will be transformed, some will be passed as is. More details on the limitations of the emulation are explained below.
+* The legacy SSE emulation is an additional drain on resources. It will consume more resources than the "native" 2.x SSE API.
+
+# Premises of legacy SSE emulation
+
+Currently the only possible emulation is the V1 SSE API. Enabling V1 SSE api emulation requires setting `emulate_legacy_sse_apis` to `["V1"]`, like:
+```
+[sse_server]
+(...)
+emulate_legacy_sse_apis = ["V1"]
+(...)
+```
+
+This will expose three additional sse endpoints:
+* `/events/sigs` -> publishes `ApiVersion`, `BlockAdded`, `DeployProcessed`, `DeployExpired`, `Fault` and `Shutdown`
+* `/events/deploys`-> publishes `ApiVersion`, `TransactionAccepted` and `Shutdown` 
+* `/events/main` -> publishes `ApiVersion`, `FinalitySignature` and `Shutdown` events
+
+Those endpoints will emit events in the same format as the V1 SSE API of the casper node. There are limitations to what Casper Sidecar can and will do, here is a list of mapping assumptions:
+
+## Translating `ApiVersion` event
+
+Legacy SSE event will be the same
+
+## Translating `BlockAdded` event
+    * When the 2.x event emits a V1 block it will be unwrapped and passed as a legacy BlockAdded, for instance a 2.x event like this:
```

Comment:
Currently the document renders in the _most likely_ unintended way, like so:

![image](https://github.com/casper-network/casper-sidecar/assets/88321181/caee1317-db5a-45ad-ad59-7180e43059bc)

```suggestion
* When the 2.x event emits a V1 block it will be unwrapped and passed as a legacy BlockAdded, for instance a 2.x event like this:
```

It could be worth checking the rendered version and see if all backticks are placed as they should.

Reply:
:+1: